### PR TITLE
doc: add best practices

### PR DESCRIPTION
**Pull request by xiang90**
/cc @hongchaodeng @colhom @joshix

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ etcd operator manages etcd clusters atop [Kubernetes][k8s-home], automating thei
 - [Backup and restore a cluster](#disaster-recovery)
 - [Rolling upgrade](#try-upgrade-etcd-cluster)
 
+Read [Best Practice](./doc/best_practices.md) can help to utilize etcd operator more effectively.
```
Comment (hongchaodeng) [inline README.md]: Read ... for more info on how to best utilize etcd operator.

File: doc/best_practices.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Best practices
+
+## Large-scale deployment
+
+To run etcd cluster at large-scale, it is important to assign etcd pods to nodes with desired resources, such as SSD, high performance network. 
+
+### Assign to nodes with desired resources
+
+Kuberentes nodes can be attached with labels. Users can [assign pods to nodes with given labels](http://kubernetes.io/docs/user-guide/node-selection/). Similarly for the etcd-operator, users can specify `Node Selector` in the cluster spec to select nodes that etcd pods will be assigned to. For example, users can label a set of nodes with SSD with label `"disk"="SSD"`. To assign etcd pods to these node, users can specify `"disk"="SSD"` in the cluster spec.
```
Comment (hongchaodeng) [inline doc/best_practices.md]: "to select nodes that etcd pods will be assigned to" ->
"to select nodes for etcd pods".

Comment (hongchaodeng) [inline doc/best_practices.md]: "users can specify ... " ->
"users can specify node selector"

File: doc/best_practices.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+### Assign to dedicated node (work in progress)
+
+Even with container isolation, not all resources are isolated and performance interference can still happen and affect etcd clusters' performance significantly. Running etcd on dedicated nodes can solve this problem.
```
Comment (hongchaodeng) [inline doc/best_practices.md]: "not all resources are isolated ... significantly" ->
"not all resources are isolated. Thus, performance ... affect etcd clusters' performance unexpectedly."

Comment (hongchaodeng) [inline doc/best_practices.md]: "Running etcd on dedicated nodes can solve this problem." ->
"We recommend dedicating nodes for each etcd server to get the best performance."

**Comment by xiang90** [comment]
@hongchaodeng All fixed.

**Comment by hongchaodeng** [comment]
LGTM.
Leave it to others for a final look.
Will merge it by tonight.

**Pull request merged**